src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,373 @@
+        final DatabaseConnection database = new DatabaseConnection(
+                this.optionAndArgumentDelegate.getOptionArgument(DATABASE_URL_INPUT).get());
+        final Connection databaseConnection = database.getConnection();
+        final String inputPath = this.optionAndArgumentDelegate.getOptionArgument(FLAG_PATH_INPUT)
+                .get();
+        this.timestamp = new Timestamp(Instant.now().toEpochMilli());
+
+        new File(inputPath).listFilesRecursively().parallelStream().forEach(file ->
+        {
+            // If this file is something we handle, read and upload the tasks contained within
+            final Optional<OutputFileType> optionalHandledFileType = getOptionalOutputType(file);
+            optionalHandledFileType.ifPresent(outputFileType ->
+            {
+
+                try (BufferedReader reader = this.getReader(file, outputFileType))
+                {
+
+                    final PreparedStatement flagSqlStatement = databaseConnection.prepareStatement(
+                            "INSERT INTO flag(flag_id, check_name, instructions, date_created) VALUES (?,?,?,?);");
+
+                    final PreparedStatement featureSqlStatement = databaseConnection
+                            .prepareStatement(String.format(
+                                    "INSERT INTO feature (flag_id, geom, osm_id, atlas_id, iso_country_code, tags, item_type, date_created) VALUES (?,%s,?,?,?,?);",
+                                    "ST_GeomFromGeoJSON(?), ?, ?"));
+
+                    final List<String> lines = reader.lines().collect(Collectors.toList());
+                    int startIndex = 0;
+                    int endIndex;
+
+                    do
+                    {
+                        endIndex = Math.min(startIndex + BATCH_SIZE, lines.size());
+                        final List<String> batchLines = lines.subList(startIndex, endIndex);
+
+                        batchLines.forEach(line ->
+                        {
+                            final JsonObject parsedFlag = new JsonParser().parse(line)
+                                    .getAsJsonObject();
+                            final JsonArray features = this.filterOutPointsFromGeojson(
+                                    parsedFlag.get(FEATURES).getAsJsonArray());
+                            final CheckFlag flag = gson.fromJson(line, CheckFlag.class);
+
+                            this.batchFlagStatement(flagSqlStatement, flag);
+
+                            StreamSupport.stream(features.spliterator(), false).forEach(
+                                    feature -> this.batchFlagFeatureStatement(featureSqlStatement,
+                                            flag, feature.getAsJsonObject()));
+                        });
+
+                        flagSqlStatement.executeBatch();
+                        featureSqlStatement.executeBatch();
+
+                        startIndex = startIndex + BATCH_SIZE;
```

Comment:
If the above ```batchLines.forEach``` statement is ```endIndex``` inclusive, then I think we'd want line to be ```startIndex = startIndex + BATCH_SIZE +1```

Reply:
I don't think so... Doing that would skip the first value after each batch.
List.sublist takes lowIndex (inclusive) and high (exclusive)

Reply:
Ah yeah if the upper bound is exclusive then the way you have it should be good